astropy/table/tests/test_operations.py

```diff
@@ -28,13 +28,23 @@
 from astropy.utils import metadata
 from astropy.utils.compat import NUMPY_LT_2_0
 from astropy.utils.compat.optional_deps import HAS_SCIPY
+from astropy.utils.masked import Masked
 from astropy.utils.metadata import MergeConflictError
 
 
 def sort_eq(list1, list2):
     return sorted(list1) == sorted(list2)
 
 
+def check_cols_equal(col1, col2):
+    """Check that col1 == col2, taking care of zero-length masked columns."""
+    assert type(col1) is type(col2) or (
+        isinstance(col1, Masked) and type(col1) is Masked(type(col2))
+    )
+    eq = np.all(col1 == col2)
+    return eq or isinstance(eq, Masked) and not eq.shape and eq.mask and eq.unmasked
```

Comment:
why `and eq.mask` ? I would expect we treat the result as an equality also in the case `eq` as a mask set to `False`

Reply:
Without the mask, just `eq` would have passed already, but, really, silly to check `Masked` behaviour here.

Note that I'm not sure I like that that `np.all(Masked([]))`  returns `Masked(True, mask=True)`, which as bool equals False. `np.all(np.ma.MaskedArray([]))` returns `np.True_`... Anyway, another issue...